taipy/core/_orchestrator/_dispatcher/_standalone_job_dispatcher.py

```diff
@@ -43,8 +43,8 @@ def _dispatch(self, job: Job):
         future = self._executor.submit(_TaskFunctionWrapper(job.id, job.task), config_as_string=config_as_string)
 
         self._set_dispatched_processes(job.id, future)  # type: ignore
-        future.add_done_callback(self._release_worker)
         future.add_done_callback(partial(self._update_job_status_from_future, job))
+        future.add_done_callback(self._release_worker)
```

Comment:
Can you give the motivation for this change? Why not release the worker before the job status update? It is done in the main process anyway.
I would release the worker first, no?

Reply:
- First is to be consistent with the same function in enterprise (https://github.com/Avaiga/taipy-enterprise/blob/develop/src/taipy/enterprise/core/_orchestrator/_dispatcher/_standalone_job_dispatcher.py#L89)
- Second, I think it just makes sense to release the worker after everything is finished, that includes updating the job status

Reply:
Sounds good.

Reply:
Actually, I changed my mind.
It is better to release the process as soon as possible (when it is available) so it can run a new job sooner. In terms of performance, it should globally reduce the waiting time for a job to be handled by a process.
What do you think?

Reply:
well if we release it before updating the status, won't new process being released create potential multi-threading issues? (not that we don't already have some)